client/my-sites/domains/index.js

```diff
@@ -192,7 +192,7 @@ export default function() {
 			siteSelection,
 			domainsController.domainsAddHeader,
 			domainsController.redirectToAddMappingIfVipSite(),
-			jetPackWarning,
+			domainsController.jetpackNoDomainsWarning,
```

Comment:
On a route like `/domains/add`, without a `:site` parameter, the `jetpackNoDomainsWarning` handler is always noop.

The `siteSelection` handler won't find any site slug in the path, so it will call `setAllSitesSelected`. Then `selectedSite` is always `null`, so `jetpackNoDomainsWarning` will always only call `next()`.

Maybe there is some gotcha I don't see though...

Reply:
Hmm, not sure if I follow what's the issue — `/domains/add`:
- shows site list when you have multiple sites, 
- if you have only one jetpack site, it chooses it and shows `jetpackNoDomainsWarning`
- if you have only one wp site, it chooses it and shows add-domains page
- if you have no sites it tells you gotta get a site :-) 

Feel free to elaborate and I can address in a separate PR.

Reply:
Welcome back @simison 🙂 

If you have only one Jetpack site, the `siteSelection` handler immediately redirects you from `/domains/add` to `/domains/add/:domain` and the `jetpackNoDomainsWarning` handler is never reached. It's only executed after the redirect, as part of the `/domains/add/:domain` handler chain.

That's why the site-less routes `/domains/add`, `/domains/add/mapping`, `/domains/add/transfer` and `/domains/add/site-redirect` don't need the `jetpackNoDomainsWarning` handler. Does that make sense?

There is no incorrect behavior, just redundant code that's never reached.